### DISCUSSION TITLE
Are there baseline CPU and Memory requirements for self-hosting standalone Next.js?

### DISCUSSION THREAD

**Discussion by jplwood** [vercel/next.js#65908] [Help]
### Summary

When self-hosting Next.js, as described in [Deploying: Self-Hosting](https://nextjs.org/docs/app/building-your-application/deploying#self-hosting), are there baseline expectations for CPU and Memory for the hosting environment? 

My app's hosting env was defaulting to `.25vCPU` and `.5gb` memory. The app was incredibly slow (TTFB consistently >2s). I bumped it way up just for testing and suddenly the app's TTFB 75th percentile is at ~500ms. That said, I'm curious if there is an optimal minimum to apply to each container to support Nextjs at baseline?

### Additional information

I'm running Next.js in standalone mode in AWS ECS Fargate, but this question applies to any self hosting.

**Comment by rootz491** [comment]
I'm also wondering about the same. Are there any recommended specs?
Also has anybody done any benchmarks on nextjs self-hosted yet? I couldn't find anything really.

**Comment by greenwookez** [comment]
What I've observed so far - I have a basic website with NextJS + PayloadCMS under the hood packed into a Docker container. The website itself isn't too large, I mean 10-15 static pages + 5 dynamic ones. And it's still in development phase, so no real load yet, just occasional QAs, so...

It feels very hard on RAM, while barely putting any load on CPU - which is pretty straightforward.
I stay on `1vCPU 3.3Ghz`, only because my IaaS provider doesn't allow less. I think I'd go with less CPU if I could.
As of RAM, I'm currently on `2gb`, and it shows that `0.6gb` is used and `1.7gb` is used with cache.

So I think I'd probably need to upgrade to somewhere near `4gb` before going into production.
